anybox/recipe/odoo/runtime/session.py

```diff
@@ -181,6 +181,7 @@ def open(self, db=None, with_demo=False):
         self.init_environments()
         self.context = self.registry('res.users').context_get(
             self.cr, self.uid)
+        self.env.context = self.context
```

Comment:
But then, a subsequent call to `init_environment()` not going through this would start afresh with `{}`
This is probably nitpicking, since you may consider `init_environment()` to be internal API, but I'd prefer it to forward self.context if available.